packages/mui-system/src/styleFunctionSx/styleFunctionSx.js

```diff
@@ -17,31 +19,69 @@ function callIfFn(maybeFn, arg) {
 }
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
-export function unstable_createStyleFunctionSx(styleFunctionMapping = defaultStyleFunctionMapping) {
-  const propToStyleFunction = Object.keys(styleFunctionMapping).reduce((acc, styleFnName) => {
-    styleFunctionMapping[styleFnName].filterProps.forEach((propName) => {
-      acc[propName] = styleFunctionMapping[styleFnName];
-    });
-
-    return acc;
-  }, {});
-
-  function getThemeValue(prop, value, theme) {
-    const inputProps = {
-      [prop]: value,
+export function unstable_createStyleFunctionSx() {
+  function getThemeValue(prop, val, theme, config) {
+    const props = {
+      [prop]: val,
       theme,
     };
 
-    const styleFunction = propToStyleFunction[prop];
-    return styleFunction ? styleFunction(inputProps) : { [prop]: value };
+    const options = config[prop];
+
+    if (!options) {
+      return { [prop]: val };
+    }
+
+    const { cssProperty = prop, themeKey, transform, style } = options;
+
+    if (val == null) {
+      return null;
+    }
+
+    const themeMapping = getPath(theme, themeKey) || {};
+
+    if (style) {
+      return style(props);
+    }
+
+    const styleFromPropValue = (propValueFinal) => {
+      let value = getValue(themeMapping, transform, propValueFinal);
+
+      if (propValueFinal === value && typeof propValueFinal === 'string') {
+        // Haven't found value
+        value = getValue(
+          themeMapping,
+          transform,
+          `${prop}${propValueFinal === 'default' ? '' : capitalize(propValueFinal)}`,
+          propValueFinal,
+        );
+      }
+
+      if (cssProperty === false) {
+        return value;
+      }
+
+      return {
+        [cssProperty]: value,
+      };
+    };
+
+    return handleBreakpoints(props, val, styleFromPropValue);
   }
 
   function styleFunctionSx(props) {
     const { sx, theme = {} } = props || {};
+
     if (!sx) {
       return null; // Emotion & styled-components will neglect null
     }
 
+    let config = defaultSxConfig;
+
+    if (theme.unstable_sxConfig) {
+      config = theme.unstable_sxConfig;
+    }
```

Comment:
Would a nullish coalescing be better?

```suggestion
    let config = theme.unstable_sxConfig ?? defaultSxConfig;
```